packages/fetchai/connections/p2p_noise/publickey.py

```diff
@@ -0,0 +1,71 @@
+import os
+import sys
+from typing import Optional
+
+import nacl.signing
+import nacl.encoding
+
+
+class PubKey:
+    def __init__(
+        self,
+        *,
+        strkey: Optional[str] = None,
+        naclkey: Optional[nacl.signing.VerifyKey] = None
+    ):
+        if naclkey is not None:
+            self._ed25519_pub = naclkey
+        elif strkey is not None:
+            self._ed25519_pub = nacl.signing.VerifyKey(
+                strkey, encoder=nacl.encoding.HexEncoder
+            )
+        else:
+            raise ValueError("Either 'strkey' or 'naclkey' must be set")
+
+    def __str__(self):
+        return self._ed25519_pub.encode(encoder=nacl.encoding.HexEncoder).decode(
+            "ascii"
+        )
+
+
+class PrivKey:
+    def __init__(self, key: Optional[str] = None):
+        if key is None:
+            self._ed25519 = nacl.signing.SigningKey.generate()
+        else:
+            self._ed25519 = nacl.signing.SigningKey(
+                key, encoder=nacl.encoding.HexEncoder
+            )
+
+    def __str__(self):
+        return self._ed25519.encode(encoder=nacl.encoding.HexEncoder).decode("ascii")
+
+    def hex(self):
+        return self._ed25519.encode(encoder=nacl.encoding.HexEncoder).decode("ascii")
+
+    def pub(self) -> PubKey:
+        return PubKey(naclkey=self._ed25519.verify_key)
+
+
+if __name__ == "__main__":
+    if len(sys.argv) < 3:
+        print("Usage: {} <swarm> <number>".format(sys.argv[0]), file=sys.stderr)
+        sys.exit(-1)
```

Comment:
this is OK. However, I'm not a fan of parsing argument by hand.

Check this for an example using `argparse`: https://github.com/fetchai/agents-aea/blob/master/scripts/parse_main_dependencies_from_lock.py#L28-L41

Docs of the standard library: https://docs.python.org/3/library/argparse.html

Reply:
file deleted